barretenberg/cpp/src/barretenberg/dsl/acir_format/ecdsa_constraints.cpp

```diff
@@ -74,8 +74,8 @@ void create_ecdsa_verify_constraints(typename Curve::Builder& builder,
     byte_array_ct pub_y_bytes = fields_to_bytes(builder, pub_y_fields);
     byte_array_ct r = fields_to_bytes(builder, r_fields);
     byte_array_ct s = fields_to_bytes(builder, s_fields);
-    bool_ct result = static_cast<bool_ct>(result_field); // Constructor enforces result = 0 or 1
-    bool_ct predicate;
+    bool_ct result = static_cast<bool_ct>(result_field);       // Constructor enforces result = 0 or 1
+    bool_ct predicate = static_cast<bool_ct>(predicate_field); // Constructor enforces predicate = 0 or 1
```

Comment:
Setting this here for two reasons: (1) it feels more consistent to set everything at once, but (2) (much more importantly) this ensures that the `predicate` is set correctly in the `signature_result.assert_equal` below. Prior to this change, it would only be set if `!input.predicate.is_constant`. This means if predicate was constant true, the local variable `predicate` would be default initialized to `false` and we would always perform the trivial `signature_result.assert_equal(signature_result);`! Yikes.

Reply:
Note this was introduced by my removal of the conditional logic in the result assertion logic. The old version didn't use `predicate` at all in the constant branch